blaze-core/src/main/scala/org/http4s/blazecore/util/ChunkWriter.scala

```diff
@@ -49,13 +50,10 @@ private[util] object ChunkWriter {
         ByteBuffer.wrap(rr.result.getBytes(ISO_8859_1))
       } else ChunkEndBuffer
     }
-    unsafeRunAsync(f) {
-      case Right(buffer) =>
-        IO { promise.completeWith(pipe.channelWrite(buffer).map(Function.const(false))); () }
-      case Left(t) =>
-        IO { promise.failure(t); () }
-    }
-    promise.future
+    val result = f
+      .flatMap(buffer => F.blocking(pipe.channelWrite(buffer)))
+      .as(false)
+    D.unsafeToFuture(result)
```

Comment:
Is it ok to stay in `Async` and to transform to `Future` only at the end?

Reply:
why is that in `blocking`? that would cause a hop and hop back to/from the blocking pool. Also, `channelWrite` is a `Future`, so that's something odd there, you want `fromFuture`

> Is it ok to stay in Async and to transform to Future only at the end?

they end up running in different thread pools (old and new diff), but I don't think that makes a huge difference. Transforming from `IO` to `Future` and back a couple times seems pointless though.

Why not
```
f.unsafeToFuture.flatMap(pipe).as(false)
```

Reply:
Good catch. I thought `pipe.channelWrite(buffer)`  was a blocking I/O operation.

Reply:
https://github.com/http4s/http4s/pull/3894/commits/61c196efb9dede4454d59d5e3956afc4a543a096